git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,620 @@
+
+You could think of each commit not as a checkpoint, but the instructions on how to get to the next checkpoint. The entire repository can be built by following a series of instructions (commits) in order. Similarly you can determine the precise state of an earlier version by following the instructions backwards.
+
+![The three white boxes are shown with a large green arrow pointing from each box to the box above it. The top box is labeled "Last committed version of repository" and the arrow pointing to it from the middle box is labeled "HEAD." The second box is labeled "Next to last version of repository" and the arrow pointing to it from the box below is labeled "HEAD~1." The bottom box is labeled "Previous version of repository" and the arrow pointing to it is labeled "HEAD~2." There are three dots under the boxes, and another three dots under the arrows indicating that this pattern continues.](media/Commits_are_pointers.svg)
+</div>
+
+## Comparing files to prior commits
+
+In order to have some more commits to compare, let's add a few more lines to Dracula's notes about Mars.
+
+First add a line to `mars.txt` so that it reads:
+
+```
+Cold and dry, but everything is my favorite color
+The two moons may be a problem for Wolfman
+But the Mummy will appreciate the lack of humidity
+```
+Let's add and commit this change:
+
+```
+$ git add mars.txt
+$ git commit -m "Discuss concerns about Mars' climate for Mummy"
+```
+
+Next let's add one more line to `mars.txt` so that it reads:
+
+```
+Cold and dry, but everything is my favorite color
+The two moons may be a problem for Wolfman
+But the Mummy will appreciate the lack of humidity
+An ill-considered change
+```
+
+Notice that we did NOT add or commit that last change to `mars.txt`. When you are working on a large project, it is very easy to lose track in your head of when you last committed, or what changes you have made since. In this section we will see how to ask Git to tell us that information.
+
+<div class = "learnmore">
+You can change things in the `planets` directory that you downloaded from GitHub!
+
+Because these files now exist on your computer, you can change them and commit those changes. The author of the new commits will be you, not Vlad Dracula. You can use `git log` to see which of you made which commit.
+
+This is a preview of how Git lets you work with collaborators to create a project together.
+</div>
+
+### Using `diff`
+
+You might have noticed that `diff` was the first word after your commit message when you entered `git show HEAD`. The lines after that are **all** the changes that you made with that commit. When comparing earlier versions with the current version, you are usually only going to want to look at one file at  a time. By typing `git diff HEAD` you are asking to see all the differences between the current state of your tracked files and the most recently committed version. Let's give it a try:
```

Comment:
I'm not sure why HEAD is getting so much attention (from Carpentries I mean) because it can move!  Maybe toward the top you note that "checkout" is a HEAD mover, and do a checkout to make sure HEAD is on main?  This will help understand why HEAD is important. https://work.njae.me.uk/2020/06/23/git-for-beginners-part-3-moving-head/